src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -654,6 +655,15 @@ export class VegaParser {
         this._setDefaultValue(defaultColor, 'config', 'trail', 'fill');
       }
     }
+
+    // provide right colors for light and dark themes
+    this._setDefaultValue(euiThemeVars.euiTextColor, 'config', 'title', 'color');
+    this._setDefaultValue(euiThemeVars.euiTextColor, 'config', 'style', 'guide-label', 'fill');
+    this._setDefaultValue(euiThemeVars.euiTextColor, 'config', 'style', 'guide-title', 'fill');
+    this._setDefaultValue(euiThemeVars.euiTextColor, 'config', 'axis', 'tickColor');
+    this._setDefaultValue(euiThemeVars.euiTextColor, 'config', 'axis', 'domainColor');
+    this._setDefaultValue(euiThemeVars.euiColorDarkShade, 'config', 'axis', 'gridColor');
+    this._setDefaultValue('transparent', 'config', 'background');
```

Comment:
I've checked locally and with this configuration, the gridline is too pronounced and makes the chart hard to read as you can see from the following screenshots
This PR
<img width="1328" alt="Screenshot 2020-08-04 at 19 09 40" src="https://user-images.githubusercontent.com/1421091/89324108-0883b600-d687-11ea-83de-a22a7d449ad6.png">
On 7.8
<img width="1332" alt="Screenshot 2020-08-04 at 19 18 22" src="https://user-images.githubusercontent.com/1421091/89324298-5c8e9a80-d687-11ea-8915-fc043ec011aa.png">


I can suggest to check and reuse the colors that EUI already have defined for the `@elastic/charts` theme: https://github.com/elastic/eui/blob/cd5a7008c4f62b7b49eac80eae13f3de7b929116/src/themes/charts/themes.ts#L124-L165 

If I'm not wrong you could map the defaults to these colors:
'config', 'title', 'color'. -> axisTitleStyle
'config', 'style', 'guide-label', 'fill' -> tickLabelStyle
'config', 'style', 'guide-title', 'fill' -> axisTitleStyle
'config', 'axis', 'tickColor' -> tickLineStyle
'config', 'axis', 'domainColor' ->  axisLineStyle
'config', 'axis', 'gridColor' ->  gridLineStyle